nixos/modules/services/backup/burp.nix

```diff
@@ -0,0 +1,451 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  libDir = "/var/lib/burp";
+
+  cfg = config.services.burp;
+  burp_conf = pkgs.writeText "burp.conf" ''
+    mode = client
+    pidfile = /run/burp/burp.pid
+
+    port = ${toString cfg.client.port}
```

Comment:
Already mentioned this in https://github.com/NixOS/nixpkgs/pull/48762#issuecomment-513591625, but by now we should really be using the approach from NixOS/rfcs#42 for this, see also https://github.com/NixOS/nixpkgs/issues/144575. By now there are a lot of NixOS modules which make use of that approach which you can use for reference. Grep for `freeformType` in nixpkgs to see them. You can also check out [the documentation](https://nixos.org/manual/nixos/stable/index.html#sec-freeform-modules).